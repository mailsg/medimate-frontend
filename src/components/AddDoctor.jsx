import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import {
  addDoctorAsync,
  fetchSpecializationsAsync,
} from '../redux/adddoctorSlice';
import styles from '../css/reserve-form.module.css';

const AddDoctor = () => {
  const dispatch = useDispatch();

  const navigate = useNavigate();

  const [doctorInfo, setDoctorInfo] = useState({
    name: '',
    time_available_from: '',
    time_available_to: '',
    bio: '',
    fee_per_appointment: '',
    specialization_id: '',
    image: '',
    location: '',
  });

  const [error, setError] = useState(null);
  useEffect(() => {
    dispatch(fetchSpecializationsAsync());
  }, [dispatch]);

  const specializations = useSelector((state) => state.app.specializations);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setDoctorInfo({ ...doctorInfo, [name]: value });
  };

  const handleAddDoctor = async () => {
    try {
      await dispatch(addDoctorAsync(doctorInfo));
      setDoctorInfo({
        name: '',
        time_available_from: '',
        time_available_to: '',
        bio: '',
        fee_per_appointment: '',
        specialization_id: '',
        image: '',
        location: '',
      });

      toast.success('Doctor added successfully!', {
        position: toast.POSITION.TOP_CENTER,
      });

      navigate('/');
    } catch (err) {
      setError(err.message || 'Error adding doctor. Please try again.');
    }
  };

  return (
    <div className={styles['reservation-container']}>
      <h1 className={styles['reserve-form-header']}>Add Doctor</h1>
      <div className={styles.addDoctorForm}>
        {error && <p className={styles.errorMessage}>{error}</p>}
        <form className={styles['add-doctor-form']}>
          <div className={[styles['flex-item'], styles.flex1].join(' ')}>
            <input
              type="text"
              placeholder="Name"
              name="name"
              value={doctorInfo.name}
              onChange={handleChange}
              className={styles.addDoctorInput}
            />
            <input
              type="text"
              placeholder="Available from"
              name="time_available_from"
              value={doctorInfo.time_available_from}
              onChange={handleChange}
              className={styles.addDoctorInput}
            />
            <input
              type="text"
              placeholder="Available to"
              name="time_available_to"
              value={doctorInfo.time_available_to}
              onChange={handleChange}
              className={styles.addDoctorInput}
            />
            <textarea
              placeholder="Short bio..."
              name="bio"
              value={doctorInfo.bio}
              onChange={handleChange}
              className={styles.addDoctorInput}
            />
          </div>
          <div className={styles['flex-item']}>
            <input
              type="text"
              placeholder="Fee per appointment"
              name="fee_per_appointment"
              value={doctorInfo.fee_per_appointment}
              onChange={handleChange}
              className={styles.addDoctorInput}
            />
            <select
              placeholder="Specialization"
              name="specialization_id"
              value={doctorInfo.specialization_id}
              onChange={handleChange}
              className={styles['select-elm']}
            >
              <option value="">Select Specialization</option>
              {specializations.map((spec) => (
                <option key={spec.id} value={spec.id}>
                  {spec.name}
                </option>
              ))}
            </select>
            <input
              type="text"
              placeholder="Image URL"
              name="image"
              value={doctorInfo.image}
              onChange={handleChange}
              className={styles.addDoctorInput}
            />
            <input
              type="text"
              placeholder="City"
              name="location"
              value={doctorInfo.location}
              onChange={handleChange}
              className={styles.addDoctorInput}
            />
            <button
              type="button"
              onClick={handleAddDoctor}
              className={styles['submit-button']}
            >
              Add Doctor
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default AddDoctor;
