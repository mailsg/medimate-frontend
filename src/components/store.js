import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './redux/doctorSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
  },
});

export default store;
