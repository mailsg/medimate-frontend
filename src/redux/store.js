import { configureStore } from '@reduxjs/toolkit';
import AppointmentSlice from './appointmentSlice';
import DoctorSlice from './doctorSlice';
import appReducer from './adddoctorSlice';

const store = configureStore({
  reducer: {
    doctor: DoctorSlice,
    appointment: AppointmentSlice,
    app: appReducer,
  },
});

export default store;
