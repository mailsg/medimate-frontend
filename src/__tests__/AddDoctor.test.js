import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddDoctor from '../components/AddDoctor';
import store from '../redux/store';

describe('Add Doctor Form Component', () => {
  test('renders Add Doctor Form component', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <AddDoctor />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
