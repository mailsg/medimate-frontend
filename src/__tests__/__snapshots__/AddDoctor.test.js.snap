// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Doctor Form Component renders Add Doctor Form component 1`] = `
<div>
  <div
    class="reservation-container"
  >
    <h1
      class="reserve-form-header"
    >
      Add Doctor
    </h1>
    <div
      class="addDoctorForm"
    >
      <form
        class="add-doctor-form"
      >
        <div
          class="flex-item flex1"
        >
          <input
            class="addDoctorInput"
            name="name"
            placeholder="Name"
            type="text"
            value=""
          />
          <input
            class="addDoctorInput"
            name="time_available_from"
            placeholder="Available from"
            type="text"
            value=""
          />
          <input
            class="addDoctorInput"
            name="time_available_to"
            placeholder="Available to"
            type="text"
            value=""
          />
          <textarea
            class="addDoctorInput"
            name="bio"
            placeholder="Short bio..."
          />
        </div>
        <div
          class="flex-item"
        >
          <input
            class="addDoctorInput"
            name="fee_per_appointment"
            placeholder="Fee per appointment"
            type="text"
            value=""
          />
          <select
            class="select-elm"
            name="specialization_id"
            placeholder="Specialization"
          >
            <option
              value=""
            >
              Select Specialization
            </option>
          </select>
          <input
            class="addDoctorInput"
            name="image"
            placeholder="Image URL"
            type="text"
            value=""
          />
          <input
            class="addDoctorInput"
            name="location"
            placeholder="City"
            type="text"
            value=""
          />
          <button
            class="submit-button"
            type="button"
          >
            Add Doctor
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
